feature/interests/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/interests/InterestsViewModel.kt

```diff
@@ -19,47 +19,43 @@ package com.google.samples.apps.nowinandroid.feature.interests
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import androidx.navigation.toRoute
 import com.google.samples.apps.nowinandroid.core.data.repository.UserDataRepository
 import com.google.samples.apps.nowinandroid.core.domain.GetFollowableTopicsUseCase
 import com.google.samples.apps.nowinandroid.core.domain.TopicSortField
 import com.google.samples.apps.nowinandroid.core.model.data.FollowableTopic
-import com.google.samples.apps.nowinandroid.feature.interests.navigation.TOPIC_ID_ARG
+import com.google.samples.apps.nowinandroid.feature.interests.navigation.InterestsDestination
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class InterestsViewModel @Inject constructor(
-    private val savedStateHandle: SavedStateHandle,
+    savedStateHandle: SavedStateHandle,
     val userDataRepository: UserDataRepository,
     getFollowableTopics: GetFollowableTopicsUseCase,
 ) : ViewModel() {
 
-    val selectedTopicId: StateFlow<String?> = savedStateHandle.getStateFlow(TOPIC_ID_ARG, null)
+    private val interestsDestination: InterestsDestination = savedStateHandle.toRoute()
 
-    val uiState: StateFlow<InterestsUiState> = combine(
-        selectedTopicId,
-        getFollowableTopics(sortBy = TopicSortField.NAME),
-        InterestsUiState::Interests,
-    ).stateIn(
-        scope = viewModelScope,
-        started = SharingStarted.WhileSubscribed(5_000),
-        initialValue = InterestsUiState.Loading,
-    )
+    val uiState: StateFlow<InterestsUiState> =
+        getFollowableTopics(sortBy = TopicSortField.NAME).map { topics ->
+            InterestsUiState.Interests(interestsDestination.topicId, topics)
+        }.stateIn(
+            scope = viewModelScope,
+            started = SharingStarted.WhileSubscribed(5_000),
+            initialValue = InterestsUiState.Loading,
+        )
 
     fun followTopic(followedTopicId: String, followed: Boolean) {
         viewModelScope.launch {
             userDataRepository.setTopicIdFollowed(followedTopicId, followed)
         }
     }
-
-    fun onTopicClick(topicId: String?) {
```

Comment:
I don't think this is safe to remove?

This function was updating the selected topic id, so that we know which one to highlight in the list.

Now, the value that is passed through will always be the one from the initial `InterestsDestination`, so I'd expect that when showing both the list and detail together, the highlighted topic won't change properly.

Reply:
You are correct, thanks for spotting this. I had mistakenly assumed that the list did not need to know the currently selected topic id. 

I've reintroduced the `Interests2PanelViewModel` to manage the state for the `InterestsScreen`. The initial topic id is obtained from `InterestsDestination` via `toRoute` and subsequent changes to the selected topic are stored using the previous mechanism of writing to `savedStateHandle`. Not ideal, but the best we could come up with for now.